# https://src.fedoraproject.org/rpms/ltrace/tree/rawhide

output.c: In function 'frame_callback':
output.c:660:37: error: '%s' directive argument is null [-Werror=format-overflow=]
  660 |         fprintf(options.output, " > %s(%s+0x%" PRIx64 ") [%" PRIx64 "]\n",
      |                                     ^~
---

--- a/expr.c
+++ b/expr.c
@@ -189,10 +189,10 @@ int
 expr_clone(struct expr_node *retp, const struct expr_node *node)
 {
 	*retp = *node;
+	struct expr_node *nlhs;
+	struct expr_node *nrhs = NULL;
 
 	switch (node->kind) {
-		struct expr_node *nlhs;
-		struct expr_node *nrhs;
 
 	case EXPR_OP_ARGNO:
 	case EXPR_OP_SELF:
--- a/output.c
+++ b/output.c
@@ -654,6 +654,9 @@ frame_callback (Dwfl_Frame *state, void
 					   NULL, NULL, NULL);
 		symname = dwfl_module_addrinfo(mod, pc, &off, &sym,
 					       NULL, NULL, NULL);
+	} else {
+		modname = "unknown";
+		symname = "unknown";
 	}
 
 	/* This mimics the output produced by libunwind below.  */
